pkg/cmd/roachtest/cluster.go

```diff
@@ -2143,6 +2169,13 @@ func (c *clusterImpl) configureClusterSettingOptions(
 	// Panic on span use-after-Finish, so we catch such bugs.
 	setUnlessExists("COCKROACH_CRASH_ON_SPAN_USE_AFTER_FINISH", true)
 
+	if c.t.(*testImpl).metamorphicConstantsEnabled {
+		// Metamorphic constants are enabled by default with the cockroach-ea binary,
+		// but we disable metamorphic constants in config.DefaultEnvVar. If metamorphic
+		// constants should be enabled, we unset the env var.
+		settings.Env = append(settings.Env, "COCKROACH_INTERNAL_DISABLE_METAMORPHIC_TESTING=false")
```

Comment:
This is not unsetting the variable, but overwriting it on the assumption that `settings.Env` are serialized in order when the underlying command is generated, right?